### DISCUSSION TITLE
Can't get Z axis to print dimensionally correct. Help!

### DISCUSSION THREAD

**Discussion by tym0t0r** [supermerill/SuperSlicer#3719] [General]
I cannot figure out how to calibrate or adjust my settings to get my prints to come out dimensionally correct along the Z axis. I'm hoping someone can point me to a SuperSlicer setting that can help me solve this. Here is an export of my slicer settings: [Z axis too high config.txt](https://github.com/supermerill/SuperSlicer/files/10952751/Z.axis.too.high.config.txt). Here are additional details about my currents setup:

**Printer**
Artillery Sidewinder X2
I've flashed an updated firmware to Marlin 2.1
I'm using 0.2mm layer height currently. I've tried with 0.3mm and gotten similar results.
Limited hardware mods (and I don't imagine they'd be a cause, but listing for completeness):

-[Z-Axis Stabiliser Brackets](https://www.thingiverse.com/thing:5320520)
-Bi-metal heat break
-0.6mm nozzle

**Material**
eSun PLA+

**Calibration Steps Taken**
I found and like Michael's [Teaching Tech](https://teachingtechyt.github.io/calibration.html) calibration pages and videos and have done my best to follow and complete most of them. These seem like the most relevant ones he touches on:

- Extruder E-steps Calibration - done
- Slicer Flow Calibration - done
- XYZ steps Calibration - done

Still not getting the results, I happened upon SuperSlicer's own calibration menu and steps. Some of them were repeats of Michael's but I went through them anyways. All good until I get to the last check, the dimensional cubes. This is where I am seeing the consistent +0.3mm variation (when measuring 20 and 30mm cubes that I print). Just to be clear the X and Y axis/sides are measuring at the expected values (or within a smaller margin of error that I'm ok with). The explanation from the calibration menu states:

> You have to print two cubes, with different sizes. If the dimensional inaccuracy scale with the size of the cubes, then it's the steps/mm of your steppers that needs some adjustments... If it doesn't scale, you can correct it by adjusting your xy compensation.

The problem is the suggested fix described only addresses X and Y compensation. There doesn't seem to be one for Z.

**Other Searching**
Most references I've been able to find only speak to prints coming out too short. From the [Slic3r Manual](https://manual.slic3r.org/troubleshooting/dimension-errors) (which I understand to be the underpinnings of SuperSlicer):

> If your vertical dimensions are wrong (i.e. along the Z axis) -and your object is usually shorter than expected- it means your nozzle is too low, thus the first layer is pressed too much on the print bed. To fix this, you might want to raise your Z endstop or increase the Z offset option in Slic3r.

If the inverse is true, then my nozzle is too high...? This can't be. I've leveled my bed and zeroed my Z settings with a feeler gauge. I can't go any lower without going into my bed--certainly not .3mm lower.
___
I imagine the solution is going to be something stupidly simple, but I just cannot figure it out; my Google-fu is failing me on this one.

**Comment by Bumblebiber** [comment]
Have you calibrated your first layer and flow rate? Your first layer might be thicker than it should be or you've got too much material on the top surface.

Print one layer and measure the thickness. Also take a look to your top surfaces if there's excess material. If so, you can tune down your total flow or just top layer flow.

**Reply by tym0t0r** [reply]
I've done both the flow calibration that SuperSlicer has within its calibration menu as well as the method spelled out by Michael on his Teaching Tech site. For the first layer I've gone through the bed leveling exercises of both sources, but both seem to be focused on getting the nozzle an appropriate height from the table for that first layer. I have not measured the thickness of a single layer print; probably a good idea to do a few test prints that are essentially 1, 2, 3, etc. layers each and measure. I'll try that tonight.

I'll take a closer look again, but I don't think there is too much material on the top layer. I've tried different settings for number of top layers and such, but I haven't seen any changes.

Thanks for the ideas. I'll let you know how tonight's testing goes.

**Reply by Bumblebiber** [reply]
Yea, measure your first layer first. If you've set your slicer to 0.2 but it's 0.4 in reality, then you've discovered where the first 0.2 inaccuracy come from. You then should set your offset 0.2 lower but also decrease first layer flow by the same percentage. 

If your flow is calibrated perfectly, all the infill layers (first layer, bottom, top, top surface infill) should feel almost flawlessly when you touch them with your finger. It may be you still have too much flow, in that case it gets worse layer by layer. In SuperSlicer, you can decrease the flowrates of each of those infill types individually. 

I, for example, have a perfectly calibrated extruder but print the top surface layer with only 85% flow.

**Reply by tym0t0r** [reply]
Did a number of test prints last night. I had the first layer height setting set at 0.2mm; the same as other layers. When I printed a single layer square, it came out measuring thicker. I adjusted the setting to 0.1mm, and now the single layers are coming out at ~0.16mm or so. Certainly an improvement.

I tried adjusting the first layer flow. I had been running it at 105%. I dropped it down to 90% thinking this might thin out the first layer even more, but unfortunately there just wasn't enough filament coming out for it to stick to the bed. I tried 95% with similar results, so I've currently got it set at 100% when I got that most recent 0.16mm measurement.

>You then should set your offset 0.2 lower but also decrease first layer flow by the same percentage.

I've been afraid to mess with Z offset (currently at 0) because I've already got my nozzle super close to the bed and I didn't want the printer trying to push it into the bed. If I'm measuring 0.4mm thick layers with a 0.2mm setting, but then set my Z offset to -0.2mm the math works, but is there a worry of physical interference between the nozzle and bed?

On one hand getting below 0.2mm for a first layer seems like a step in the right direction. On the other hand, you would hope to not have to trick the slicer into producing what you want. Regardless, I'll take it for now.

I also intended to reduce my top layer flow as you suggested, but for some strange reason I started getting issues with layer shift when I tried to print a few calibration cubes, and I ran out of time last night.

So, homework for tonight seems to be potentially tweaking Z offset, maybe playing with different combinations of first layer flow and layer height, and also testing different top layer flow settings. Sound about right? Any other suggestions?

I very much appreciate your time and input!!

**Reply by Bumblebiber** [reply]
Your first layer height in the slicer should always stay unchanged. 

If you've set it to 0.2, but your real first layer is 0.4 then your z-offset is 0.2mm too high. Just tune it down by this amount in the printer menu by that value. No reason to worry.

After that you will notice too much filament, because your first layer flow was tuned to this wrong offset. Just print anything (a mid to large size print) and tune your flow on the run. Once it looks good you can cancel this print and write this percentage in First Layer Flow in Superslicer.

Print one layer again and repeat this process until you're satisfied with the result. Note that there are still some inconsistencies in your buildplate, so values +-0.02mm should be okay. 

Your first layer then should come out super crisp so that you don't want to stop stroking it :)

If solid layers 2 and 3 over- or underextruded then you can adjust the filament flowrate in filament settings. But if you make any change here, you also need to adjust the first layer flow rate again.


Another thing to check: Is your filament diameter consistent? Measure it at 5 different points and write the average into filament diameter box (in filament settings).

**Reply by tym0t0r** [reply]
>Just tune it down by this amount in the printer menu by that value.

I definitely don't have that much room to go down on the printer. When doing bed leveling and setting my Z, I use a 0.05mm feeler gauge instead of a sheet of paper. Once there is friction with the feeler gauge, I remove it and lower the Z by 0.0025 (the smallest increment my printer can do), and then save it. This is where my concern of additional Z offset within SuperSlicer comes in. The math would indicate I should set it at -0.2mm, but based on my feeler gauge and Z setting, that would be through the bed. Maybe I'm misunderstanding what you are proposing, but I don't see how what you suggest will work out...?

I'll check the filament diameter.

**Reply by Bumblebiber** [reply]
Before you do your z-offset calibration, what z position do you move the printhead to? 

Go to z=0.2 and then do your feeler gauge stuff, but with a 0.2mm one of course :) 

When your printer prints the first layer, it's never on z=0.

**Reply by diemen1245** [reply]
Texture of build plate can also affect your Z dimension. if you use a powder coated PEI build plate, filling of those valleys can be the difference of 0.3mm.

**Reply by Fumanseano13** [reply]
Hey check under printer settings tab in SS, it may have a field marked z full step= 0.00625 or similar. If that is enabled try setting that value as 0 and see if that helps
